Userland/Libraries/LibManual/Node.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2022, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "Node.h"
+#include "PageNode.h"
+#include "SectionNode.h"
+#include <AK/LexicalPath.h>
+#include <AK/Optional.h>
+#include <AK/StringView.h>
+#include <LibCore/Stream.h>
+#include <LibManual/Path.h>
+
+namespace Manual {
+
+ErrorOr<NonnullRefPtr<PageNode>> Node::try_create_from_query(Vector<StringView, 2> const& query_parameters)
+{
+    if (query_parameters.size() > 2)
+        return Error::from_string_literal("Queries longer than 2 strings are not supported yet");
+
+    auto query_parameter_iterator = query_parameters.begin();
+
+    if (query_parameter_iterator.is_end())
+        return Error::from_string_literal("No query given");
```

Comment:
This feels like a programming mistake, not a runtime one, to pass an empty query vector to this method. if we know there's no query, why even bother calling the method?